[MRG] Updates to reports

madprime opened the pull request:
* adds button for symptom report to homepage
* if checkins are on, display management form on another page
* add page listing past symptom reports

Potential/likely TODOs remaining:
* a "no symptom" button on the main page (interrelated to next item & notes below)
* (maybe?) better handling of no symptom reports (see notes below)
* less crappy "report history"? report history on front page?

Also thinking about:
* adding ability to create historic symptom report records (probably requires adding a field to the model, update/new form)
* JSON data for symptom reports stored on Open Humans
* JSON data for symptom reports shared as public data

## no symptom reporting

I broke [the rules](https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html) and made a token-bearing GET request that modifies data on the server (logging a "no symptom" report when the page loads). I think this might be creating extra entries when the page is accidentally reloaded – so – somehow this should be cleaned up.

Options…
1. require an extra button press to confirm once the page loads (POST)
2. do something smart (don't re-use a token more than once for "no symptom" reports, don't save more than once over some period of time, something like this)

gedankenstuecke commented on reports/templates/reports/list.html:
```diff
@@ -0,0 +1,43 @@
+          {% for symptom in report.symptoms.all %}
+          {{ symptom.label }}{% if not forloop.last %}, {% endif %}
+          {% endfor %}
+        {% endif %}
```
This currently only displays the symptoms that aren't fever. Which looks weird if you do a report that's having fever but no other symptoms (A completely empty report will show "none", an "only fever"-one will be just empty).

gedankenstuecke commented on quantified_flu/templates/quantified_flu/home.html:
```diff
@@ -63,7 +63,34 @@ <h2>Data Sources</h2>
+    <p>
+      <b>Note: "no symptoms" reports are also important!</b> By tracking if and when you have
+      symptoms of illness, we can develop better models to understand this data.
+    </p>
```
I noticed that currently it's impossible to file an empty symptom report through the web front–end as the `fever` entry is required. Is a true no-symptom report different from one that's done through the front-end at this point? My comment re: display of symptom reports below suggests that they are different, as the display for 'empty, select no fever' is different from 'empty, reported through email'.

madprime replied:
There _is_ a specific field marking "no symptoms" report. So yeah, I want to improve this – probably to make a button at the top that simply says "report no symptoms".

I think it's also correct (if implicit) to suggest requiring the fever field to be done is unnecessarily onerous – whether symptoms are reported or not. If someone isn't trying to change that field (to explicitly say "no fever")… "no fever" is almost certainly the case. We should at least drop that requirement, I think. (We can still save it differently, but I expect we'll interpret it this way.)

Option 1: separate front-page link for "report no symptoms"
Option 2: extend symptom report with a quick button for "report no symptoms"

I'm leaning to option 2. Less clutter. (As we extend things, clutter will accumulate.)

But if we do it that way, do we have different forms for email check-in vs. website? Or switch to a single link in the check-in email (plus additional click to finish a "no symptom" report)? It might seem minor, but I think consistency is important for UX.

gedankenstuecke replied:
Having the button on top sounds good to me! 👍

madprime commented:
Updates (rebased)…

- added quick "no symptom report" button to the symptom report if it's accessed via site (not email token)
- added token field to symptom report model, attach these if reporting via token
- redirect user to report history after making symptom report
- add fever guess to the symptom report history
- made fever guess option in symptom report form (i.e. it's all optional now)
- makes symptom report list reverse chronological and somewhat prettier
- added ability to view other user symptoms reports, if public data (currently this isn't exposed; you have to guess the URL)
- added JSON data for these report list URLs (personal & other user reports)
